tests/trainer/logging/test_eval_loop_logging_1_0.py

```diff
@@ -321,3 +321,62 @@ def test_monitor_val_epoch_end(tmpdir):
         checkpoint_callback=checkpoint_callback,
     )
     trainer.fit(model)
+
+def test_log_works_in_validation_callback(tmpdir):
+    """
+    Tests that log can be called within callback
+    """
+    import pytorch_lightning as pl
+
+    os.environ['PL_DEV_DEBUG'] = '1'
+
+    funcs_name = [f for f in dir(callbacks.Callback) if ("val" in f)]
+
+    def logic_func(x):
+        func_idx, func_name, args = x
+        return f"""trainer = locals().get('trainer')\n  try:\n      trainer.get_model().log('{func_name}', {func_idx})\n  except Exception as e:\n        print('{func_name}', e)""" \
+            if "trainer" in args else "pass"
+    
+    scripted_callback = create_scriptable_callback(funcs_name, logic_func)
+
+    loss_values, patience, expected_stop_epoch = ([6, 5, 5, 5, 5, 5], 3, 4)
+
+    class TestModel(BoringModel):
+
+        validation_return_values = torch.Tensor(loss_values)
+        count = 0
+
+        def validation_step(self, batch, batch_idx):
+            output = self.layer(batch)
+            loss = self.loss(batch, output)
+            self.log('c', self.count)
+            return {"x": loss}
+
+        def validation_epoch_end(self, outputs):
+
+            loss = self.validation_return_values[self.count]
+            self.count += 1
+            self.log("test_val_loss", loss)
+            return {"test_val_loss": loss}
+
+    max_epochs = 5
+    model = TestModel()
+    # using Early Stopping to make sure callback_metrics is correctly set for `on_validation_epoch_end` hook
+    early_stop_callback = pl.callbacks.EarlyStopping(monitor="test_val_loss", patience=3, verbose=True)
```

Comment:
doesn't make sense to early stop on a test split haha. that's cheating ;)

Reply:
You are right ! Changed it :)